lib/compiler.js

```diff
@@ -12,7 +12,13 @@ const AV = require("./assignment-visitor.js");
 const assignmentVisitor = new AV;
 
 const shebangRegExp = /^#!.*/;
-const importExportRegExp = /\b(?:im|ex)port\b/;
+
+// Matches any import or export identifier as long as it's not preceded by
+// a `.` character (to avoid matching module.export, for example). Because
+// Reify replaces all import and export declarations with module.import
+// and module.export calls, this logic should prevent the compiler from
+// ever having to recompile code it has already compiled.
+const importExportRegExp = /(?:^|[^.])\b(?:im|ex)port\b/;
```

Comment:
This makes the steps ~x3 more for a failed match or match of just exports. Maybe a combo of a light and then a heavier regexp... looking into it.